### PR TITLE
Add an environment variable $CHRUBY_SET_GEM_HOME to disable setting $GEM_HOME

### PR DESCRIPTION
**Pull request by eregon**
This is a proof-of-concept PR to address #422.
It still needs documentation and a changelog entry.

@postmodern @havenwood What do you think?

### CODE REVIEW

**Comment by eregon** [comment]
@postmodern @havenwood Could we merge this? What do you think?

File: share/chruby/chruby.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ function chruby_reset()
 	PATH=":$PATH:"; PATH="${PATH//:$RUBY_ROOT\/bin:/:}"
 	[[ -n "$GEM_ROOT" ]] && PATH="${PATH//:$GEM_ROOT\/bin:/:}"
 
-	if (( UID != 0 )); then
+	if (( UID != 0 )) && [ "$CHRUBY_SET_GEM_HOME" != "false" ]; then
```
Comment (havenwood) [inline share/chruby/chruby.sh]: Other tests expressions use double brackets, which I think would be best to stick with for consistent compatibility and style.

My first impulse was to prefer `0` over `"false"` for falsey, but I don't have a strong defense.

Comment (eregon) [reply share/chruby/chruby.sh]: OK, I'll switch to double brackets.

I think `false` is a bit more readable as in:
```bash
# In e.g., ~/.profile
export CHRUBY_SET_GEM_HOME=false

# Looks more cryptic to me:
export CHRUBY_SET_GEM_HOME=0
```

Comment (eregon) [reply share/chruby/chruby.sh]: Done, please re-review.

**Comment by havenwood** [comment]
This looks good to me.

The only question I would have is whether to check if the variable is set rather than `"false"` or `0`.  This seems like as good a solution as any to me, unless @postmodern has an opinion on it. It seems there are some shell libs (like [bash-boolean-helpers](https://github.com/Jeff-Russ/bash-boolean-helpers)) but hardly worth adopting here.

**Comment by postmodern** [comment]
Posted some thoughts: https://github.com/postmodern/chruby/issues/422#issuecomment-557379476

I've seen people prefer testing for `"true"` instead of not `"false"`.

**Comment by eregon** [comment]
> Posted some thoughts: [#422 (comment)](https://github.com/postmodern/chruby/issues/422#issuecomment-557379476)

Thanks, I replied there.

> I've seen people prefer testing for `"true"` instead of not `"false"`.

`CHRUBY_SET_GEM_HOME` has a default of `true`, so an explicit `false` seems the clearest to me.

**Comment by eregon** [comment]
I made a PR which doesn't need user configuration and just does the "right thing" based on whether the default gem home is writable: https://github.com/postmodern/chruby/pull/431

It seems better than this PR, so I'll close this.

**Pull request closed without merging**